spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -0,0 +1,625 @@
+    public void shouldDeleteChildrenPoolsWhenParentPoolIsDeleted() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+
+        ProductDTO product = Products.randomSKU()
+            .addAttributesItem(this.buildAttribute("virt_limit", "10"));
+
+        product = adminClient.ownerProducts()
+            .createProductByOwner(owner.getKey(), product);
+
+        PoolDTO pool = this.createPool(adminClient, owner, product);
+
+        // Fetch the org's pools, verify the target pool and its unmapped guest pool exist
+        List<PoolDTO> pools = adminClient.pools()
+            .listPools(owner.getId(), null, null, true, null);
+
+        assertNotNull(pools);
+        assertEquals(2, pools.size());
+
+        Optional<PoolDTO> nfilter = pools.stream()
+            .filter(p -> "NORMAL".equals(p.getType()))
+            .findFirst();
+
+        assertTrue(nfilter.isPresent());
+        PoolDTO npool = nfilter.get();
+
+        Optional<PoolDTO> dfilter = pools.stream()
+            .filter(p -> !"NORMAL".equalsIgnoreCase(p.getType()))
+            .findFirst();
+
+        assertTrue(dfilter.isPresent());
+        PoolDTO dpool = dfilter.get();
+
+        // Delete the pool
+        adminClient.pools().deletePool(pool.getId());
+
+        // Verify the both pools were deleted
+        pools = adminClient.pools()
+            .listPools(owner.getId(), null, null, true, null);
+
+        assertNotNull(pools);
+        assertEquals(0, pools.size());
+
+        assertNotFound(() -> adminClient.pools().getPool(npool.getId(), null, null));
+        assertNotFound(() -> adminClient.pools().getPool(dpool.getId(), null, null));
+    }
+
+    @Test
+    @DisplayName("should include calculated attributes when fetching pools")
+    public void shouldIncludeCalculatedAttributesWhenFetchingPools() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        PoolDTO pool = this.createPool(adminClient, owner);
+
+        ApiClient ownerClient = this.createUserClient(adminClient, owner);
+
+        String consumerName = StringUtil.randomSuffix("test_consumer");
+        ConsumerDTO consumer = this.createConsumer(ownerClient, owner, consumerName, null);
+        ApiClient consumerClient = this.createConsumerClient(ownerClient, consumer);
+
+        PoolDTO fetched = consumerClient.pools().getPool(pool.getId(), consumer.getUuid(), null);
+        assertNotNull(fetched);
+
+        Map<String, String> attributes = fetched.getCalculatedAttributes();
+        assertNotNull(attributes);
+
+        assertThat(attributes)
+            .isNotNull()
+            .hasSizeGreaterThanOrEqualTo(2)
+            .containsEntry("suggested_quantity", "1")
+            .containsEntry("compliance_type", "Standard");
+    }
+
+  // it 'should allow fetching content delivery network by pool id' do
```

Comment:
is the ruby test comment needed here, if the java test is already commented below, and the ruby test still exists in the .rb file?

Reply:
done